serenata_toolbox/federal_senate/federal_senate_dataset.py

```diff
@@ -0,0 +1,100 @@
+
+        for filename in filenames:
+            file_path = os.path.join(self.path, filename)
+            data = pd.read_csv(file_path, encoding = "utf-8")
+            dataset = pd.concat([dataset, data])
+
+        dataset['date'] = pd.to_datetime(dataset['date'], errors='coerce')
+        dataset['cnpj_cpf'] = dataset['cnpj_cpf'].str.replace(r'\D', '')
+
+        dataset.to_csv(reimbursement_path, compression='xz', index=False, encoding='utf-8')
+
+        return reimbursement_path
+
+    def __translate_file(self, csv_path):
+        output_file_path = csv_path.replace('.csv', '.xz')
+
+        data = pd.read_csv(csv_path,
+                           sep=';',
+                           encoding = "ISO-8859-1",
+                           skiprows=1)
+
+        data.columns = map(str.lower, data.columns)
+
+        data.rename(columns={
+            'ano': 'year',
+            'mes': 'month',
+            'senador': 'congressperson_name',
+            'tipo_despesa': 'expense_type',
+            'cnpj_cpf': 'cnpj_cpf',
+            'fornecedor': 'supplier',
+            'documento': 'document_id',
+            'data': 'date',
+            'detalhamento': 'expense_details',
+            'valor_reembolsado': 'reimbursement_value',
+        }, inplace=True)
+
+        data['expense_type'] = data['expense_type'].astype('category')
+
+        data['expense_type'] = \
+            data['expense_type'].astype('category')
+
+        categories = {
+            'Aluguel de imóveis para escritório político, compreendendo despesas concernentes a eles.':
+                'Rent of real estate for political office, comprising expenses concerning them',
+            'Aquisição de material de consumo para uso no escritório político, inclusive aquisição ou locação de software, despesas postais, aquisição de publicações, locação de móveis e de equipamentos. ':
+                'Acquisition of consumables for use in the political office, including acquisition or leasing of software, postal expenses, acquisition of publications, rental of furniture and equipment',
+            'Contratação de consultorias, assessorias, pesquisas, trabalhos técnicos e outros serviços de apoio ao exercício do mandato parlamentar':
+                'Recruitment of consultancies, advisory services, research, technical work and other services in support of the exercise of the parliamentary mandate',
+            'Divulgação da atividade parlamentar':
+                'Publicity of parliamentary activity',
+            'Locomoção, hospedagem, alimentação, combustíveis e lubrificantes':
+                'Locomotion, lodging, food, fuels and lubricants',
+            'Passagens aéreas, aquáticas e terrestres nacionais':
+                'National air, water and land transport',
+            'Serviços de Segurança Privada':
+                'Private Security Services'
+        }
+
+        categories = [categories[cat] for cat in data['expense_type'].cat.categories]
```

Comment:
At this line looks that is a translation from portuguese to english but it's changing state of categories variable here. It is making this code more complex than it could be. :/

May it could be done in a function just to give it a name and help with clarity.
 Show all reviewers